app_dart/lib/src/request_handlers/push_benchmark_to_center.dart

```diff
@@ -0,0 +1,144 @@
+import 'dart:async';
+
+import 'package:appengine/appengine.dart';
+import 'package:gcloud/db.dart';
+import 'package:meta/meta.dart';
+import 'package:metrics_center/flutter.dart' as mc;
+
+import '../datastore/cocoon_config.dart';
+import '../foundation/providers.dart';
+import '../model/appengine/time_series.dart';
+import '../model/appengine/time_series_value.dart';
+import '../request_handling/body.dart';
+import '../request_handling/request_handler.dart';
+import '../service/datastore.dart';
+
+/// Pushes devicelab benchmark results to Metrics Center storage.
+///
+/// Metrics Center aggregates benchmark data in the unified [MetricPoint] format
+/// and sends them to performance dashboards.
+@immutable
+class PushBenchmarkToCenter extends RequestHandler<Body> {
+  PushBenchmarkToCenter(
+    Config config, {
+    @visibleForTesting Logging logger,
+    @visibleForTesting DatastoreService datastoreService,
+    @visibleForTesting FlutterDestinationProvider flutterDestinationProvider,
+  })  : logger = logger ?? Providers.serviceScopeLogger(),
+        datastoreService =
+            datastoreService ?? DatastoreService.defaultProvider(config.db),
+        flutterDestinationProvider =
+            flutterDestinationProvider ?? defaultFlutterDestinationProvider,
+        super(config: config);
+
+  final Logging logger;
+  final DatastoreService datastoreService;
+  final FlutterDestinationProvider flutterDestinationProvider;
+
+  @override
+  Future<Body> get() async {
+    final int sinceTimeMillis =
+        parseSinceTimeMillis(request.uri.queryParameters);
+    if (sinceTimeMillis == null) {
+      logger.error('The sinceTimeMillis parameter is missing');
+      return Body.empty;
+    }
+
+    final List<TimeSeriesValue> timeSeriesValues =
+        await readTimeSeriesValue(sinceTimeMillis);
+    final Map<String, TimeSeries> timeSeriesMap = await readTimeSeriesMap();
+
+    final List<BenchmarkMetricPoint> points =
+        await transform(timeSeriesMap, timeSeriesValues);
+    logger.debug('Transformed ${points.length} metric points.');
+    logger.debug('The first one is ${points.first}');
+
+    final mc.FlutterDestination destination =
+        await flutterDestinationProvider(config);
+    await destination.update(points);
+    return Body.empty;
+  }
+
+  @visibleForTesting
+  int parseSinceTimeMillis(Map<String, String> queryParameters) {
+    final String s = queryParameters['sinceTimeMillis'] ?? '';
+    return int.tryParse(s);
+  }
+
+  /// Returns a map containing all time series keyed by its id.
+  Future<Map<String, TimeSeries>> readTimeSeriesMap() async {
+    final Map<String, TimeSeries> map = <String, TimeSeries>{};
+    await for (TimeSeries ts in datastoreService.db.query<TimeSeries>().run()) {
+      // The id is string typed, see the defination of [TimeSeries].
+      map[ts.id as String] = ts;
+    }
+    return map;
+  }
+
+  /// Reads benchmark records with [createTimestamp] later than [sinceMillis].
```

Comment:
Maybe change "later than" to "strictly later than" to clarify that we're doing ">" instead of ">=" comparison.